utils/filepattern-generator-plugin/tests/test_main.py

```diff
@@ -0,0 +1,53 @@
+
+from pathlib import Path
+import os, sys
+dirpath = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(dirpath, '../'))
+import unittest
+from src.main import *
+import json
+
+inpDir = Path(dirpath).parent.joinpath('images')
+outDir = Path(dirpath).parent.joinpath('out')
+pattern='p0{r}_x{x+}_y{y+}_wx{t}_wy{p}_c{c}.ome.tif'
+chunkSize=9
+filename='pattern_generator'
+data = {'p00_x01_y{rr}_wx0_wy0_c{t}.ome.tif':30, 
+        'p00_x01_y{rr}_wx0_wy1_c{t}.ome.tif':30,
+        'p00_x01_y{rr}_wx0_wy2_c{t}.ome.tif':30
+        }
+
+class Test_Filepattern_Generator(unittest.TestCase):
+
+    def setUp(self) -> None:
+
+        self.inpDir = inpDir
+        self.pattern = pattern
+        self.chunkSize = chunkSize
+        self.filename=filename
+        self.outDir=outDir
+        self.data = data
+
+    # def test_get_grouping(self):
+    #     groupBy=None
+    #     groupby, count = get_grouping(self.data, self.pattern, groupBy, self.chunkSize)
+    #     self.assertFalse(count!= self.chunkSize)
+    #     self.assertFalse(groupby== None)
+    #     groupBy='t'
+    #     groupby, count = get_grouping(self.data, self.pattern, groupBy, self.chunkSize)
+    #     self.assertTrue(count== self.chunkSize)
+    #     self.assertTrue(groupby!= None)
```

Comment:
Either uncomment, remove, or add a comment to explain why this is being kept in the code as a comment.